com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/Fog.cs

```diff
@@ -200,4 +222,71 @@ public sealed class FogColorParameter : VolumeParameter<FogColorMode>
         public FogColorParameter(FogColorMode value, bool overrideState = false)
             : base(value, overrideState) { }
     }
+
+    /// <summary>
+    /// Options that control the quality and resource intensity of the volumetric fog.
+    /// </summary>
+    public enum FogControl
+    {
+        /// <summary>
+        /// Use this mode if you want to change the fog control properties based on a higher abstraction level centered around performance.
+        /// </summary>
+        Balance,
+
+        /// <summary>
+        /// Use this mode if you want to have direct access to the internal properties that control volumetric fog.
+        /// </summary>
+        Manual
+    }
+
+    /// <summary>
+    /// A <see cref="VolumeParameter"/> that holds a <see cref="ExposureMode"/> value.
+    /// </summary>
+    [Serializable]
+    public sealed class FogControlParameter : VolumeParameter<FogControl>
+    {
+        /// <summary>
+        /// Creates a new <see cref="FogControlParameter"/> instance.
+        /// </summary>
+        /// <param name="value">The initial value to store in the parameter.</param>
+        /// <param name="overrideState">The initial override state for the parameter.</param>
+        public FogControlParameter(FogControl value, bool overrideState = false) : base(value, overrideState) { }
+    }
+
+    /// <summary>
+    /// Options that control which denoising algorithms Unity should use on the volumetric fog signal.
+    /// </summary>
+    public enum FogDenoisingMode
+    {
+        /// <summary>
+        /// Use this mode to not filter the volumetric fog.
+        /// </summary>
+        None = 0,
+        /// <summary>
+        /// Use this mode to reproject data from previous frames to denoise the signal. This is effective for static lighting, but it can lead to severe ghosting artifacts for highly dynamic lighting.
+        /// </summary>
+        Reprojection = 1 << 0,
```

Comment:
It's not really denoising. It's temporal integration.

Reply:
I know, but it is really for users to have an intuitive feeling of what to click. The term denoising is more obvious for them, filtering or integration, less.